cpp/src/arrow/testing/random_test.cc

```diff
@@ -499,6 +499,45 @@ TEST(RandomList, Basics) {
   }
 }
 
+TEST(RandomNestedBatchOf, Basics) {
```

Comment:
Can this just be part of the `RandomList` suite as a new case?

Reply:
And for all three tests, can you call out specifically that we want to test child field nullability is preserved?